packet/adaptationfield.go

```diff
@@ -0,0 +1,342 @@
+}
+
+const pcrStart = 6
+
+// opcrLength returns the length of the OPCR, if there is no OPCR then its length is zero
+func (af AdaptationField) opcrLength() int {
+	if af.HasOPCR() {
+		return 6
+	}
+	return 0
+}
+
+// opcrStart returns the start index of where the OPCR field should be
+func (af AdaptationField) opcrStart() int {
+	return pcrStart + af.pcrLength()
+}
+
+// spliceCountdownLength returns the length of the splice countdown, if there is no splice countdown then its length is zero
+func (af AdaptationField) spliceCountdownLength() int {
+	if af.HasSplicingPoint() {
+		return 1
+	}
+	return 0
+}
+
+// spliceCountdownStart returns the start index of where the splice countdown field should be
+func (af AdaptationField) spliceCountdownStart() int {
+	return pcrStart + af.pcrLength() + af.opcrLength()
+}
+
+// transportPrivateDataLength returns the length of the transport private data,
+// if there is no transport private data then its length is zero
+func (af AdaptationField) transportPrivateDataLength() int {
+	if af.HasTransportPrivateData() {
+		// cannot extend beyond adaptation field, number of bytes
+		// for field stored in transportPrivateDataLength
+		return 1 + int(af[af.transportPrivateDataStart()])
+	}
+	return 0
+}
+
+// transportPrivateDataStart returns the start index of where the transport private data should be
+func (af AdaptationField) transportPrivateDataStart() int {
+	return pcrStart + af.pcrLength() + af.opcrLength() + af.spliceCountdownLength()
+}
+
+// adaptationExtensionLength returns the length of the adaptation field extension,
+//  if there is no adaptation field extension then its length is zero
+func (af AdaptationField) adaptationExtensionLength() int {
+	if af.HasAdaptationFieldExtension() {
+		return 1 + int(af[af.adaptationExtensionStart()])
+	}
+	return 0
+}
+
+// adaptationExtensionStartreturns the length of the adaptation field extension,
+// if there is no adaptation field extension then its length is zero
+func (af AdaptationField) adaptationExtensionStart() int {
+	return pcrStart + af.pcrLength() + af.opcrLength() +
+		af.spliceCountdownLength() + af.transportPrivateDataLength()
+}
+
+// calculates the length of the Adaptation Field
+// (with respect to the start of the packet) excluding stuffing
+func (af AdaptationField) calculateMinLength() int {
+	return pcrStart +
+		af.pcrLength() + af.opcrLength() + af.spliceCountdownLength() +
+		af.transportPrivateDataLength() + af.adaptationExtensionLength()
+}
+
+func (af AdaptationField) setBit(index int, mask byte, value bool) {
+	if af.invalid() {
+		return
+	}
+	if value {
+		af[index] |= mask
+	} else {
+		af[index] &= ^mask
+	}
+}
+
+func (af AdaptationField) getBit(index int, mask byte) bool {
+	if af.invalid() {
+		return false
+	}
+	return af[index]&mask != 0
+}
+
+func (af AdaptationField) bitDelta(index int, mask byte, value bool) int {
+	oldValue := af.getBit(index, mask)
+	if value != oldValue {
+		if value {
+			return 1 // growing
```

Comment:
Instead of documenting the return values in-line, write a GoDoc comment for this function that has the return value in it.